File: flutter/flutter-sdk-bridge/ping_oidc/ios/ping_oidc/Sources/ping_oidc/PingOidcPlugin.swift
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Flutter
import UIKit

/// Registers the generated `PingOidcHostApi` Pigeon channel.
public class PingOidcPlugin: NSObject, FlutterPlugin {
    private var api: OidcHostApiImpl?

    public static func register(with registrar: FlutterPluginRegistrar) {
        let instance = PingOidcPlugin()
        let api = OidcHostApiImpl()
        instance.api = api
        PingOidcHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: api)
        registrar.publish(instance)
    }

    public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
        PingOidcHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: nil)
        api = nil
    }
}

File: flutter/flutter-sdk-bridge/ping_oidc/android/src/main/kotlin/com/pingidentity/flutter/oidc/PingOidcPlugin.kt
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.flutter.oidc

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Registers the generated [PingOidcHostApi] Pigeon channel. */
class PingOidcPlugin : FlutterPlugin {
    private var api: OidcHostApiImpl? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        api = OidcHostApiImpl()
        PingOidcHostApi.setUp(binding.binaryMessenger, api)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        PingOidcHostApi.setUp(binding.binaryMessenger, null)
        api?.shutdown()
        api = null
    }
}
